tiatoolbox/tools/patchextraction.py

```diff
@@ -280,7 +302,18 @@ def filter_coordinates_fast(
         flag_list = []
         for coord in scaled_coords:
             this_part = tissue_mask[coord[1] : coord[3], coord[0] : coord[2]]
-            flag_list.append(np.any(this_part > 0))
+            patch_area = np.prod(this_part.shape)
+            pos_area = this_part.sum()
+            if min_mask_ratio == 0:
```

Comment:
I'm not sure why you considered an special case `min_mask_ratio == 0`.
I believe the whole situation can be rewritten with one `if` statement:
```
if pos_area > patch_area * min_mask_ratio: # note that ">" is used instead of ">="
    flag_list.append(True)
else:
    flag_list.append(False)
```

hopefully, this would also help with coverage and you won't need to add more tests to cover this bit.

Reply:
@mostafajahanifar The issue I found was, when setting min_mask_ratio=1. With that value I would expect to include all patches completely inside the mask. However, if we only use `>` no patch will be included because the maximum value is 1.

Reply:
Good point about all white mask patches 👍 
Your suggestion would do the job, however, something like below can be more readable which takes into account the special case of `min_mask_ratio=1`:
```
if (pos_area  == patch_area) or (pos_area > patch_area * min_mask_ratio):
    flag_list.append(True)
else:
    flag_list.append(False)
```

Reply:
I was checking this test:
https://github.com/rogertrullo/tiatoolbox/blob/0ee90054fc4c4b88251be57b868f51d32554ab69/tests/test_patch_extraction.py#L414-L425
```
    bbox_list = np.array(
        [
            [0, 0, 4, 4],
            [0, 4, 4, 8],
            [4, 0, 8, 4],
            [4, 4, 8, 8],
            [8, 0, 12, 4],
            [8, 4, 12, 8],
        ]
    )
    mask = np.zeros([9, 6])
    mask[0:4, 3:8] = 1  # will flag first 2
```
which after clipping gives this:
```
array([[0, 0, 4, 4],
       [0, 4, 4, 8],
       [4, 0, 6, 4],
       [4, 4, 6, 8],
       [6, 0, 6, 4],
       [6, 4, 6, 8]])
```
which would give the last two patches to be of area=0, and if we add min_area_ratio=0, they will be added incorrectly. Should we add a special condition on area==0?